File: auth/src/main/java/com/eirsteir/coffeewithme/authservice/service/RoleService.java
package com.eirsteir.coffeewithme.authservice.service;

import com.eirsteir.coffeewithme.authservice.domain.Role;
import com.eirsteir.coffeewithme.authservice.domain.RoleType;


public interface RoleService {

    Role getOrCreateRole(RoleType type);

}

File: auth-service/src/main/kotlin/com/eirsteir/coffeewithme/authservice/service/RoleService.kt
package com.eirsteir.coffeewithme.authservice.service

import com.eirsteir.coffeewithme.authservice.domain.Role
import com.eirsteir.coffeewithme.authservice.domain.RoleType

interface RoleService {
    fun getOrCreateRole(type: RoleType): Role
}